tests/test_base_resources.py

```diff
@@ -534,14 +531,14 @@ def test_upload_data_chunking_callbacks(self, mock_bar_request):
         # Set chunking to 3 bytes per chunk
         # Upload file
         f.upload_data(file_obj, chunk_size=1,
-                      upload_callback=self.upload_callback,
-                      commit_callback=self.commit_callback)
+                      upload_callback=upload_callback,
+                      commit_callback=commit_callback)
 
         assert reqmock.call_count == 7
-        assert self.upload_callback_count == 6
-        assert self.upload_callback_progress == 100.0
-        assert self.commit_callback_count == 1
-        assert self.commit_callback_progress == 100.0
```

Comment:
you may have forgotten to delete the upload_callback and commit_callback functions

you can then do the progress assertions by using the mock.assert_called_with() function
upload_callback.assert_called_with(100.0)

it acts on the most recent call, but the other calls are available for inspection using the upload_callback.mock_calls property if you need them